src/main/java/edu/ucsb/cs56/mapache_search/controllers/HomePageController.java

```diff
@@ -0,0 +1,116 @@
+import edu.ucsb.cs56.mapache_search.entities.SearchResultEntity;
+import edu.ucsb.cs56.mapache_search.entities.UserVote;
+import edu.ucsb.cs56.mapache_search.membership.AuthControllerAdvice;
+import edu.ucsb.cs56.mapache_search.entities.AppUser;
+import edu.ucsb.cs56.mapache_search.search.SearchObject;
+import edu.ucsb.cs56.mapache_search.search.SearchParameters;
+import edu.ucsb.cs56.mapache_search.search.SearchResult;
+import edu.ucsb.cs56.mapache_search.search.SearchService;
+import net.minidev.json.JSONObject;
+import edu.ucsb.cs56.mapache_search.search.SearchResult;
+import java.io.IOException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.sql.Connection;
+import java.util.Date;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+@Controller
+public class HomePageController {
+
+    private Logger logger = LoggerFactory.getLogger(SearchController.class);
+
+    @Autowired
+    private SearchService searchService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private SearchResultRepository searchRepository;
+
+    @Autowired
+    private VoteRepository voteRepository;
+
+    @Autowired
+    private AuthControllerAdvice controllerAdvice;
+
+    @Autowired
+    private StackExchangeQueryService queryService;
+
+    @Autowired
+    private PreviewProviderService previewService;
+
+    @Autowired
+    public HomePageController(SearchResultRepository searchRepository) {
+        this.searchRepository = searchRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model) {
+        model.addAttribute("searchObject", new SearchObject());
+        List<UserVote> upVoteList = voteRepository.findByUpvoteOrderByTimestampDesc(true); //A List that stores UserVote only when the user upvoted 
+        ArrayList<String> upVoteLinks = new ArrayList<String>(); // A list that stores the url that got upvoted
+        //This for loop serves to get all the url linsk that have been upvoted
+        for(int pos = 0; pos < upVoteList.size(); pos++)
```

Comment:
Can you think of a way to change this line so that we limit the number of links here, rather than in the template? Hint: it involves an or expression (a ||)